contracts/utils/RetireToucanCarbon.sol

```diff
@@ -0,0 +1,1405 @@
+                poolBalance,
+                bondPrice,
+                DAO
+            );
+        }
+        return true;
+    }
+
+    function setFeeAmount(uint256 _amount) external onlyOwner returns (bool) {
+        emit FeeUpdated(feeAmount, _amount);
+
+        feeAmount = _amount;
+        return true;
+    }
+
+    function setPoolRouter(address _toucanPool, address _router)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        poolRouter[_toucanPool] = _router;
+        return true;
+    }
+
+    function addPool(
+        address _toucanPool,
+        address _router,
+        address _bondDepository
+    ) external onlyOwner returns (bool) {
+        bool result;
+
+        require(!listContains(toucanPools, _toucanPool), "Pool already added");
+
+        if (!listContains(toucanPools, _toucanPool)) {
+            toucanPools.push(_toucanPool);
+        }
+        result = !isToucanPool[_toucanPool];
+        isToucanPool[_toucanPool] = result;
+        poolRouter[_toucanPool] = _router;
+        bondDepository[_toucanPool] = _bondDepository;
+
+        emit PoolAdded(_toucanPool, result);
+        return true;
+    }
+
+    function updateBondDepository(address _toucanPool, address _bondDepository)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        require(isToucanPool[_toucanPool], "Not a Toucan Carbon Pool");
+
+        bondDepository[_toucanPool] = _bondDepository;
+        return true;
+    }
+
+    /**
+        @notice checks array to ensure against duplicate
+        @param _list address[]
+        @param _token address
+        @return bool
+     */
+    function listContains(address[] storage _list, address _token)
+        internal
+        view
+        returns (bool)
+    {
+        for (uint256 i = 0; i < _list.length; i++) {
+            if (_list[i] == _token) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function getNeededUSDC(
+        address _toucanPool,
+        address _router,
+        uint256 _poolAmount
+    ) public view returns (uint256, address[] memory) {
+        address[] memory path = new address[](2);
+
+        path[0] = USDC;
+        path[1] = _toucanPool;
+
+        uint256[] memory amountIn = IUniswapV2Router02(_router).getAmountsIn(
+            _poolAmount,
+            path
+        );
+
+        return ((amountIn[0] * 1005) / 1000, path); // Account for .5% slippage
+    }
+
+    function getNeededKLIMA(
+        address _toucanPool,
+        address _router,
+        uint256 _poolAmount
+    ) public view returns (uint256, address[] memory) {
+        address[] memory path = new address[](2);
+
+        path[0] = KLIMA;
+        path[1] = _toucanPool;
+
+        uint256[] memory amountIn = IUniswapV2Router02(_router).getAmountsIn(
+            _poolAmount,
+            path
+        );
+
+        return ((amountIn[0] * 1005) / 1000, path); // Account for .5% slippage
```

Comment:
Why .5 and not .3?

Reply:
Updated to be .1% which is the default for both Sushi and Quickswap in an future commit